pkg/assembler/backends/inmem/pkgEqual_test.go

```diff
@@ -657,6 +661,17 @@ func TestIngestPkgEquals(t *testing.T) {
 			if err != nil {
 				return
 			}
+
+			// make the expected input and output canonical
+			for _, pkgEquals := range test.ExpHE {
+				makeCanonicalPkgEquals(pkgEquals)
+			}
+			for _, pkgEquals := range got {
+				makeCanonicalPkgEquals(pkgEquals)
+			}
+			slices.SortFunc(test.ExpHE, lessPkgEquals)
+			slices.SortFunc(got, lessPkgEquals)
```

Comment:
hmm.. can we fold this into a cmpopt?

Reply:
Yes that would work, but taking order into account this way isn't consistent with how the makeCanonical functions do it. Instead, I [factored this section of code out](https://github.com/mdeicas/guac/blob/e05bdf2ed900d77dfaae1eec29993309cc2f17cf/pkg/assembler/backends/inmem/pkg_test.go#L758) to a helper function. 

Is this OK or do you still prefer to use cmpopts to sort the top level list of PkgEqual nodes?

Reply:
that's fine - maybe we can add a TODO there to see if we make a cmpopts to it. I think the `MakeCanonicalPackageSlice`  function probably should be private.